sgt-back-end/index.js

```diff
@@ -0,0 +1,124 @@
+const pg = require('pg');
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:lfz@localhost/studentGradeTable'
+});
+const express = require('express');
+const app = express();
+app.use(express.json());
+
+app.get('/api/grades', (req, res) => {
+  const sql = `
+    select *
+      from "grades"
+  `;
+
+  db.query(sql)
+    .then(result => {
+      res.status(200).json(result.rows);
+    })
+    .catch(err =>{
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occured.'
+      });
+    });
+})
+
+app.post('/api/grades', (req, res) => {
+  const newGrade = req.body;
+  if (!req.body.name || !req.body.course || !req.body.score) {
+    return (res.status(400).json({
+      error: 'Missing name, course, or score'
+    }));
+  }
+  if (req.body.score <= 0 || req.body.score > 100) {
+    return res.status(400).json({
+      error: 'Score must be a number between 1 and 100'
+    });
+  }
+  const sql = `
+    insert into "grades" ("name", "course", "score")
+    values ($1, $2, $3)
+    returning *
+  `;
+  const params = [newGrade.name, newGrade.course, newGrade.score];
+
+  db.query(sql, params)
+    .then(result => {
+      res.status(201).json(result.rows[0]);
+    })
+    .catch(err =>{
+      console.error(err);
+      return res.status(500).json({
+        error: 'An unexpected error occured.'
+      });
+    })
+})
+
+app.put('/api/grades/:gradeId', (req, res) => {
+  const gradeId = req.params.gradeId;
+  if(!gradeId){
+    return res.status(404).json({
+      error: 'The grade targeted by the Id does not exist in the database'
+    });
+  }
+  if(req.params.id <= 0 || !req.body.name || !req.body.score || !req.body.score){
+    return res.status(400).json({
+      error: 'Inputted gradeId, name, score, or course is invalid'
+    });
+  }
+  const sql = `
+    update "grades"
+      set "name" = $1, "course" = $2, "score" = $3
+      where "gradeId" = $4`;
+  const params = [req.body.name, req.body.course, req.body.score, req.params.gradeId];
+  db.query(sql, params)
+    .then(result => {
+      res.status(200).json(result.rows[0]);
```

Comment:
Here you want to check if `result.rows[0]` contains a value or not, if `result.rows[0]` is `undefined` that means the `gradeId` didn't exist so you want to send back a `404` Not Found error. If `result.rows[0]` has a value that means everything worked, send back a `200` and the grade object. Similar to what you did in your `delete` endpoint.